config/formats.ts

```diff
@@ -2,7 +2,79 @@
 // The rules that formats use are stored in data/rulesets.js
 import {Utils} from './../lib/utils';
 
-export const Formats: (FormatsData | {section: string, column?: number})[] = [
+import {SideFormats} from './custom-formats';
+
+// interface for the builder.
+interface FormatSection {
+	section: string
+	column?: number
+	formats: FormatsData[];
+}
+
+// function for merging the two lists
+function merge(main: (FormatsData | {section: string, column?: number})[],
+			   side: (FormatsData | {section: string, column?: number})[]):
+			   (FormatsData | {section: string, column?: number})[] {
+	// result that is return and makes the actual list for formats.
+	var result: (FormatsData | {section: string, column?: number})[] = [];
+
+	// used as a intermediary to build the final list.
+	var build: formatSection[] = [];
+
+	// used to track location to keep formats under their sections.
+	var loc: number = -1;
+
+	// populates the origonal sections and formats easily
+	// there should be no repeat sections at this point.
+	main.forEach(element => {
+		if (element.section) {
+			build.push({section: element.section, column: element.column, formats: []});
+			loc++;
+		} else if (element.name) {
+			build[loc].formats.push(element);
+		}
+	});
+
+	// merges the second list the hard way. Accounts for repeats.
+	side.forEach(element => {
+		// finds the section and makes it if it doesn't exist.
+		if (element.section) {
+			loc = 0;
+			var found: boolean = false;
+
+			// finds the loc of the section header (or next loc if it's new)
+			build.forEach(comp => {
+				if (comp.section === element.section) {
+					found = true;
+					return;
+				}
+				loc++;
+			});
```

Comment:
```suggestion
			for (const comp of build) {
				if (comp.section === element.section) {
					found = true;
					break;
				}
				loc++;
			}
```
Here's an example of using a `for-of` loop.